ThunderCloud/QuizQuestionContainerViewController.swift

```diff
@@ -0,0 +1,267 @@
+		
+		hintLabel.isHidden = question?.hint == nil
+		hintLabel.text = question?.hint
+	}
+	
+	override func viewDidDisappear(_ animated: Bool) {
+		super.viewDidDisappear(animated)
+		// check if the back button was pressed
+		guard isMovingFromParentViewController, let quiz = quiz else { return }
+		quiz.currentQuestion?.reset()
+		self.quiz?.currentIndex = quiz.currentIndex - 1
+	}
+	
+	private func redraw() {
+		
+		guard let question = question else { return }
+
+		questionLabel.font = UIFont.preferredFont(forTextStyle: .body).withWeight(.bold)
+		questionLabel.text = question.question
+		
+		hintLabel.font = UIFont.preferredFont(forTextStyle: .callout)
+		embeddedView.backgroundColor = ThemeManager.shared.theme.backgroundColor
+		view.backgroundColor = ThemeManager.shared.theme.backgroundColor
+		
+		navigationItem.titleView = titleView()
+	}
+	
+	open func titleView() -> UIView? {
+		
+		guard let quiz = quiz, let questions = quiz.questions else { return nil }
+		
+		// UIView to contain multiple elements for navigation bar
+		let progressContainer = UIView(frame: CGRect(x: 0, y: 0, width: 140, height: 44))
+		
+		let progressLabel = UILabel(frame: CGRect(x: 0, y: 3, width: progressContainer.bounds.width, height: 22))
+		progressLabel.textAlignment = .center
+		progressLabel.font = ThemeManager.shared.theme.boldFont(ofSize: 16)
+		progressLabel.textColor = navigationController?.navigationBar.tintColor
+		progressLabel.backgroundColor = .clear
+		
+		if StormLanguageController.shared.isRightToLeft {
+			progressLabel.text = "\(questions.count) \("of".localised(with: "_QUIZ_OF")) \(quiz.currentIndex + 1)"
+		} else {
+			progressLabel.text = "\(quiz.currentIndex + 1) \("of".localised(with: "_QUIZ_OF")) \(questions.count)"
+		}
+		
+		progressContainer.addSubview(progressLabel)
+		
+		let progressView = UIProgressView(frame: CGRect(x: 0, y: 22, width: progressContainer.bounds.width, height: 22))
+		progressView.progressTintColor = ThemeManager.shared.theme.progressTintColour
+		progressView.trackTintColor = ThemeManager.shared.theme.progressTrackTintColour
+		progressView.progress = 0
+		
+		if StormLanguageController.shared.isRightToLeft {
+			let transform = CGAffineTransform(rotationAngle: .pi)
+			progressView.transform = transform
+		}
+			
+		progressView.progress = Float(quiz.currentIndex) / Float(questions.count)
+		progressView.setY(progressView.frame.minY + 10)
+		progressView.transform = progressView.transform.concatenating(CGAffineTransform(scaleX: 1.0, y: 3.0))
+		
+		progressContainer.addSubview(progressView)
+			
+		let progressStartCap = UIView(frame: CGRect(x: progressView.frame.minX - 2, y: progressView.frame.minY, width: 6, height: 6))
+		progressStartCap.cornerRadius = 3.0
+		progressStartCap.backgroundColor = progressView.progressTintColor
+		progressContainer.addSubview(progressStartCap)
+		
+		let progressEndCap = UIView(frame: CGRect(x: progressView.frame.maxX - 3, y: progressView.frame.minY, width: 6, height: 6))
```

Comment:
Is there any reason why it's `-3` here, and `-2` above?

Reply:
Not entirely sure, it was taken from the old objective-c code...